packages/@vuepress/core/lib/client/serverEntry.js

```diff
@@ -1,14 +1,15 @@
 import { createApp } from './app'
 
 export default context => new Promise((resolve, reject) => {
-  const { app, router } = createApp(true /* isServer */)
-  const { url } = context
-  const { fullPath } = router.resolve(url).route
+  createApp(true /* isServer */).then(({ app, router }) => {
```

Comment:
Why not using async await here?

Reply:
@kefranabg Need to resolve in `router.onReady()` callback

Reply:
This should works the same, and looks easier to read imho. WDYT?

```
new Promise(async (resolve, reject) => {
const { app, router } = await createApp(true /* isServer */)
const { url } = context
const { fullPath } = router.resolve(url).route
if (fullPath !== url) {
   return reject({ url: fullPath })
}

router.push(url)
router.onReady(() => resolve(app))
```

Reply:
https://eslint.org/docs/rules/no-async-promise-executor

Although our eslint config does not open this rule, but it's a recommended rule 🤔

Reply:
OK 👍

Reply:
From the doc: `If an async executor function throws an error, the error will be lost and won't cause the newly-constructed Promise to reject. This could make it difficult to debug and handle some errors.` Indeed it make sense 😉